scripts/install_ebpf.psm1

```diff
@@ -219,48 +146,171 @@ function Install-eBPFComponents
+    Print-eBPFComponentsStatus "Verifing the status of eBPF drivers and services after the installation..." | Out-Null
 
+    # Optionally enable KMDF verifier and tag tracking.
     if ($KMDFVerifier) {
-        # Enable KMDF verifier and tag tracking.
         Enable-KMDFVerifier
     }
 
-    # Start all components.
-    Start-eBPFComponents -KmTracing $KmTracing -KmTraceType $KmTraceType
+    # Start KM tracing.
+    Start-WPRTrace -KmTracing $KmTracing -KmTraceType $KmTraceType
 }
 
-function Stop-eBPFComponents
+function Uninstall-eBPFComponents
 {
-    # Stop user mode service.
-    Stop-Service "eBPFSvc" -ErrorAction Ignore 2>&1 | Write-Log
-
-    # Stop the drivers.
+    # Firstly, uninstall the extra drivers that are not installed by the MSI package.
     $EbpfDrivers.GetEnumerator() | ForEach-Object {
-        Stop-Service $_.Name -ErrorAction Ignore 2>&1 | Write-Log
+        if (-not $_.Value.InstalledByMsi) {
+            Write-Log("Stopping $($_.Key) service...") -ForegroundColor Green
+            sc.exe stop $_.Key 2>&1 | Write-Log
+            if ($LASTEXITCODE -ne 0) {
+                Write-Log("Failed to stop $($_.Key) service.") -ForegroundColor Red
+            } else {
+                Write-Log("$($_.Key) service stopped.") -ForegroundColor Green
+            }
+            Write-Log("Deleting $($_.Key) service...") -ForegroundColor Green
+            sc.exe delete $_.Key 2>&1 | Write-Log
+            if ($LASTEXITCODE -ne 0) {
+                Write-Log("Failed to delete $($_.Key) service.") -ForegroundColor Red
+            } else {
+                Write-Log("$($_.Key) service deleted.") -ForegroundColor Green
+            }
+            $driverPath = if (Test-Path -Path ("$pwd\{0}" -f $_.Value.Name)) {
+                "$pwd\{0}" -f $_.Value.Name
+            } elseif (Test-Path -Path ("$pwd\drivers\{0}" -f $_.Value.Name)) {
+                "$pwd\drivers\{0}" -f $_.Value.Name
+            }
+            if ($null -ne $driverPath) {
+                Write-Log("Deleting driver file: $driverPath") -ForegroundColor Green
+                Remove-Item -Path $driverPath -Force -ErrorAction SilentlyContinue
+            } else {
+                Write-Log("Driver file not found for $($_.Key).") -ForegroundColor Red
+            }
+        }
     }
-}
 
-function Uninstall-eBPFComponents
-{
-    Stop-eBPFComponents
-    Unregister-eBPFComponents
-    Remove-Item "$Env:systemroot\system32\drivers\*bpf*" -Force -ErrorAction Stop 2>&1 | Write-Log
-    Remove-Item "$Env:systemroot\system32\*bpf*" -Force -ErrorAction Stop 2>&1 | Write-Log
+    # Clear export program info for the sample driver.
+    Write-Log("Running 'export_program_info_sample.exe --clear'...")
+    if (Test-Path -Path "export_program_info_sample.exe --clear") {
+        .\export_program_info_sample.exe --clear
+        if ($LASTEXITCODE -ne 0) {
+        Write-Log("Failed to run 'export_program_info_sample.exe --clear'.") -ForegroundColor Red
```

Comment:
IMO this a catastrophic error as well and we ought to abort right away.

Reply:
@dv-msft same as comment above.